timemachine/md/minimizer.py

```diff
@@ -525,7 +525,7 @@ def scipy_minimize(
     def val_and_grad_fn_bfgs(x_flattened):
         x = x_flattened.reshape(final_shape)
         u, grad_full = val_and_grad_fn(x)
-        return u, grad_full.reshape(-1).astype(np.float64)
+        return u, np.asarray(grad_full.reshape(-1)).astype(np.float64)
```

Comment:
Without `jax.config.update("jax_enable_x64", True)` this would trigger tons of errors. Also has an impact on the results of BFGS, my current hypothesis is that passing float32 to BFGS (https://github.com/scipy/scipy/blob/main/scipy/optimize/_optimize.py#L1314) is losing enough precision to produce worse results.

Reply:
Also without this change leads to failures if using `L-BFGS-B` when not enabling 64 bit.